wagtail/admin/templates/wagtailadmin/shared/side_panels/preview.html

```diff
@@ -12,18 +12,25 @@
 >
     <div class="w-preview__sizes">
         {% block size_buttons %}
-            <label class="w-preview__size-button">
-                {% icon name="mobile-alt" %}
-                <input type="radio" class="w-preview__proxy" name="preview-size" value="mobile" data-action="w-preview#togglePreviewSize" data-w-preview-target="size" data-device-width="375" data-default-size checked aria-label="{% trans 'Preview in mobile size' %}" />
-            </label>
-            <label class="w-preview__size-button">
-                {% icon name="tablet-alt" %}
-                <input type="radio" class="w-preview__proxy" name="preview-size" value="tablet" data-action="w-preview#togglePreviewSize" data-w-preview-target="size" data-device-width="768" aria-label="{% trans 'Preview in tablet size' %}" />
-            </label>
-            <label class="w-preview__size-button">
-                {% icon name="desktop" %}
-                <input type="radio" class="w-preview__proxy" name="preview-size" value="desktop" data-action="w-preview#togglePreviewSize" data-w-preview-target="size" data-device-width="1280" aria-label="{% trans 'Preview in desktop size' %}" />
-            </label>
+            {% for size in object.preview_sizes %}
+                <label class="w-preview__size-button">
+                    {% icon name=size.icon %}
+                    <input
```

Comment:
Now that this could have more than 3 items, we might need to update the CSS to be more defensive if they overflow the container

<img width="341" alt="image" src="https://github.com/user-attachments/assets/4d26d28d-7224-4adc-bb73-aa0388ac7cd8">

Reply:
Have made a commit with a few tweaks to the CSS to improve this, including when the auto update is turned off:

## Before

<img width="392" alt="image" src="https://github.com/user-attachments/assets/7947d2b6-16a4-426f-a284-39e8f5c76e2a">

## After

<img width="392" alt="image" src="https://github.com/user-attachments/assets/f008d98c-785b-4fda-a1a3-f33d63a6cf51">